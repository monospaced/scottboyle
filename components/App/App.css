@import url("../../theme/fonts.css");
@import url("../../theme/vars.css");

html {
  background: var(--background-color);
  color: var(--text-color);
  font-family: theme-font, sans-serif;
  line-height: 1.5;
  scroll-behavior: smooth;
  text-rendering: optimizeLegibilty;
}

body {
  margin: 0;
}

footer,
header,
nav,
section {
  display: block;
}

a {
  background-color: transparent;
  color: var(--link-color);
  text-decoration: none;
  text-decoration-skip: ink;
  transition: color 0.15s;
}

a:hover {
  outline: 0;
  text-decoration: underline;
}

a:active {
  color: var(--theme-color);
  outline: 0;
  text-decoration: none;
  transition: none;
}

abbr[title] {
  border-bottom: none;
  text-decoration: none;
}

h1,
h2,
h3 {
  color: var(--text-emphasis-color);
  font-weight: normal;
  margin: 1.5rem 0;
}

h1 {
  font-size: var(--great-primer);
  line-height: 1;
}

h2 {
  font-size: var(--english);
  line-height: 1;
}

h3 {
  font-size: var(--pica);
  margin-bottom: 0;
}

p {
  margin: 1.5rem 0;
}

h3 + p,
h3 + ul {
  margin-top: 0;
}

ul {
  list-style: none;
  margin: 1.5rem 0;
  padding: 0;
}

strong {
  color: var(--text-emphasis-color);
  font-weight: normal;
}

small {
  color: var(--text-color);
  font-size: var(--long-primer);
}

time {
  font-size: var(--long-primer);
  white-space: nowrap;
}

img {
  display: block;
}

.App {
  margin: 0 auto;
  max-width: 25rem;
  padding: 0 1.5rem;
}

@media (--larger-viewport) {
  .App {
    max-width: 40.5rem;
  }

  .App > div:first-child::after {
    clear: both;
    content: "";
    display: table;
  }
}
