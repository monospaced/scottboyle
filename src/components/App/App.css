@import url("../../theme/vars.css");

.App {
  margin: 0 auto;
  max-width: 25rem;
  padding: 0 1.5rem;
}

@media (--larger-viewport) {
  .App {
    max-width: 40.5rem;
  }

  .App:after {
    clear: both;
    content: "";
    display: table;
  }
}
