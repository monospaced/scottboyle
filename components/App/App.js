import React, {Component} from 'react';
import 'normalize.css';
import './App.css';
import '../Main/Main.css';
import Header from '../Header/Header';
import Nav from '../Nav/Nav';

class App extends Component {
  constructor(props) {
    super(props);
  }
  render() {
    const {data} = this.props.route;
    return (
      <div>
        <Header
          title={data.title}
          subtitle={data.subtitle}
        />
        {React.cloneElement(this.props.children, {data: data})}
        <Nav
          projects={data.projects}
          routes={this.props.routes}
        />
      </div>
    );
  }
}

module.exports = App;
