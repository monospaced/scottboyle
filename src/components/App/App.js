import React from "react";

import "./App.css";
import Header from "../Header/Header";
import Nav from "../Nav/Nav";

const App = ({ children, route: { data }, routes }) => {
  const { projects, subtitle, title } = data;

  return (
    <div className="App">
      <Header subtitle={subtitle} title={title} />
      {React.cloneElement(children, { data })}
      <Nav projects={projects} routes={routes} />
    </div>
  );
};

export default App;
