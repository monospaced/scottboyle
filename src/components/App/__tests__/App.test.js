import React from "react";
import { shallow } from "enzyme";

import App from "../App.js";

describe("App component", () => {
  it("should render correctly", () => {
    const props = {
      route: { data: { subtitle: "", title: "", projects: [] } },
      routes: [],
    };
    const component = shallow(
      <App {...props}>
        <main class="Main" />
      </App>,
    );
    expect(component).toMatchSnapshot();
  });
});
