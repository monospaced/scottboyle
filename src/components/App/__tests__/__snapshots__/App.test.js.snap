// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render correctly 1`] = `
<div
  className="App"
>
  <Header
    subtitle=""
    title=""
  />
  <main
    class="Main"
    data={
      Object {
        "projects": Array [],
        "subtitle": "",
        "title": "",
      }
    }
  />
  <Nav
    projects={Array []}
    routes={Array []}
  />
</div>
`;
